tests/storage/clone.go

```diff
@@ -213,9 +298,50 @@ var _ = SIGDescribe("[Serial]VirtualMachineClone Tests", func() {
 				Expect(vmClone.Status.RestoreName).To(BeNil())
 			})
 
+			It("simple clone with snapshot source", func() {
+				By("Creating a VM")
+				sourceVM = createVM()
+				Eventually(func() virtv1.VirtualMachinePrintableStatus {
+					sourceVM, err = virtClient.VirtualMachine(sourceVM.Namespace).Get(sourceVM.Name, &v1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					return sourceVM.Status.PrintableStatus
+				}, 30*time.Second, 1*time.Second).Should(Equal(virtv1.VirtualMachineStatusStopped))
+
+				By("Creating a snapshot from VM")
+				snapshot := createSnapshot(sourceVM)
+				snapshot = waitSnapshotContentsExist(snapshot)
+				// "waitSnapshotReady" is not used here intentionally since it's okay for a snapshot source
+				// to not be ready when creating a clone. Therefore, it's not deterministic if snapshot would actually
+				// be ready for this test or not.
+				// TODO: use snapshot's createDenyVolumeSnapshotCreateWebhook() once it's refactored to work outside
+				// of snapshot tests scope.
+
+				By("Deleting VM")
+				err = virtClient.VirtualMachine(sourceVM.Namespace).Delete(sourceVM.Name, &v1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Creating a clone with a snapshot source")
+				vmClone = generateCloneFromSnapshot(snapshot, targetVMName)
+				createCloneAndWaitForFinish(vmClone)
+
+				By(fmt.Sprintf("Getting the target VM %s", targetVMName))
+				targetVM, err = virtClient.VirtualMachine(sourceVM.Namespace).Get(targetVMName, &v1.GetOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("Making sure target is runnable")
+				targetVM = expectVMRunnable(targetVM)
+
+				By("Making sure snapshot source is not being deleted")
```

Comment:
This will add additional 30 secs to the overall test execution time. are we sure this check is necessary?

Reply:
I'm not sure :smile: 
I thought this it is crucial to make sure we don't harm the existing snapshot, but maybe it's an overkill. I've kept the check but removed the `Consistently` part :)